rare/components/tabs/downloads/__init__.py

```diff
@@ -1,298 +1,334 @@
-        self.progress_bar.setValue(0)
-        self.dl_speed.setText("n/a")
-        self.time_left.setText("n/a")
-        self.cache_used.setText("n/a")
-        self.downloaded.setText("n/a")
-        self.analysis = None
-
-    def progress_update(self, ui_update: UIUpdate):
-        self.progress_bar.setValue(
-            100 * ui_update.total_downloaded // self.analysis.dl_size
-        )
-        self.dl_speed.setText(f"{get_size(ui_update.download_compressed_speed)}/s")
-        self.cache_used.setText(
-            f"{get_size(ui_update.cache_usage) if ui_update.cache_usage > 1023 else '0KB'}"
-        )
-        self.downloaded.setText(
-            f"{get_size(ui_update.total_downloaded)} / {get_size(self.analysis.dl_size)}"
-        )
-        self.time_left.setText(self.get_time(ui_update.estimated_time_left))
-        self.signals.dl_progress.emit(
-            100 * ui_update.total_downloaded // self.analysis.dl_size
-        )
-
-    def get_time(self, seconds: Union[int, float]) -> str:
-        return str(datetime.timedelta(seconds=seconds))
-
-    def on_install_dialog_closed(self, download_item: InstallQueueItemModel):
-        if download_item:
-            self.install_game(download_item)
-            # lk: In case the download in comming from game verification/repair
-            if w := self.update_widgets.get(download_item.options.app_name):
-                w.update_button.setDisabled(True)
-                w.update_with_settings.setDisabled(True)
-            self.signals.set_main_tab_index.emit(1)
-        else:
-            if w := self.update_widgets.get(download_item.options.app_name):
-                w.update_button.setDisabled(False)
-                w.update_with_settings.setDisabled(False)
+                return
 
-    def get_install_options(self, options: InstallOptionsModel):
+        # lk: if we finished a repair, and we have a disabled update, re-enable it
+        if self.updates_group.contains(result.options.app_name):
+            self.updates_group.set_widget_enabled(result.options.app_name, True)
 
+        if result.code == DlResultCode.FINISHED and self.queue_group.count():
+            self.__start_download(self.queue_group.pop_front())
+        elif result.code == DlResultCode.STOPPED and self.__forced_item:
+            self.__start_download(self.__forced_item)
+            self.__forced_item = None
+        else:
+            self.__reset_download()
+
+    def __reset_download(self):
+        self.download_widget.setPixmap(QPixmap())
+        self.download_widget.ui.kill_button.setDisabled(True)
+        self.download_widget.ui.dl_name.setText(self.tr("No active download"))
+        self.download_widget.ui.progress_bar.setValue(0)
+        self.download_widget.ui.dl_speed.setText("n/a")
+        self.download_widget.ui.time_left.setText("n/a")
+        self.download_widget.ui.cache_used.setText("n/a")
+        self.download_widget.ui.downloaded.setText("n/a")
```

Comment:
Those labels are for values. "n/a" is a placeholder which might or might not require translation. But we could use something else like ellipsis "..." or empty to avoid translations as they provide little to no value for such labels